articles/decoding-jwt-structure.md

```diff
@@ -0,0 +1,276 @@
+# Decoding JWTs Structure
+
+By Rishi Raj Jain
+
+<insert-fig>
+
+JSON Web Tokens (JWTs) are compact, URL-safe tokens commonly used in modern authentication and authorization systems, particularly with OAuth2 flows. This guide provides a technical deep-dive into JWT architecture, examining each component and its role in the token's security and functionality.
+
+JWTs are sometimes incorrectly referred to as "JWT Tokens" - this would expand to "JSON Web Token Token," which is redundant. We'll use "JWT" throughout this guide. Additionally, while JWTs are frequently used for authentication, they're technically just base64-encoded JSON containers that can serve various purposes beyond auth. This guide focuses on their authentication use case while explaining the underlying structure.
+
+Let's examine a real JWT example. Here's a freshly generated token with line breaks added for readability (normally JWTs are single-line strings):
+
+```
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6ImY1ODg5MGQxOSJ9.eyJhdWQiO
+iI4NWEwMzg2Ny1kY2NmLTQ4ODItYWRkZS0xYTc5YWVlYzUwZGYiLCJleHAiOjE2NDQ4ODQ
+xODUsImlhdCI6MTY0NDg4MDU4NSwiaXNzIjoiYWNtZS5jb20iLCJzdWIiOiIwMDAwMDAwM
+C0wMDAwLTAwMDAtMDAwMC0wMDAwMDAwMDAwMDEiLCJqdGkiOiIzZGQ2NDM0ZC03OWE5LTR
+kMTUtOThiNS03YjUxZGJiMmNkMzEiLCJhdXRoZW50aWNhdGlvblR5cGUiOiJQQVNTV09SR
+CIsImVtYWlsIjoiYWRtaW5AZnVzaW9uYXV0aC5pbyIsImVtYWlsX3ZlcmlmaWVkIjp0cnV
+lLCJhcHBsaWNhdGlvbklkIjoiODVhMDM4NjctZGNjZi00ODgyLWFkZGUtMWE3OWFlZWM1M
+GRmIiwicm9sZXMiOlsiY2VvIl19.dee-Ke6RzR0G9avaLNRZf1GUCDfe8Zbk9L2c7yaqKME
+```
+
+While this appears as random characters, understanding JWT structure reveals the encoded data within. This guide focuses on signed JWTs (also called JWS - JSON Web Signatures), which are the most common type in production systems.
+
+A signed JWT consists of three distinct segments separated by periods (`.`):
+
+1. **Header**: Contains metadata about the token (starts with `eyJhbGc` in our example)
+2. **Payload/Body**: Contains the actual data/claims (starts with `eyJhdWQ`)
+3. **Signature**: Cryptographic proof of integrity (starts with `dee-K`)
+
+<insert-fig>
+
+Let's break down each component and understand how they work together to create a secure, self-contained token.
```

Comment:
Provide a fully decoded token. Then mention how we'll break down and explain each section